sphinx/application.py

```diff
@@ -298,10 +299,19 @@ def create_builder(self, buildername):
         if buildername is None:
             logger.info(_('No builder selected, using default: html'))
             buildername = 'html'
-        if buildername not in self.builderclasses:
-            raise SphinxError(_('Builder name %s not registered') % buildername)
-
-        builderclass = self.builderclasses[buildername]
+        if buildername in self.builderclasses:
+            builderclass = self.builderclasses[buildername]
+        else:
+            entry_points = iter_entry_points('sphinx.builders', buildername)
+            try:
+                entry_point = next(entry_points)
+            except StopIteration:
+                raise SphinxError('Builder name %s not registered or available'
+                                  ' through entry point' % buildername)
+            else:
+                builderclass = entry_point.load()
+                extension_module = sys.modules[builderclass.__module__]
+                extension_module.setup(self)
```

Comment:
Just noticed. Please use `sphinx.extensions.load_extension()` to load an extension.